hs_labels/models.py

```diff
@@ -338,6 +344,9 @@ def unsave_label(self, this_label):
         label_string = UserLabels.clean_label(this_label)
         UserStoredLabels.objects.filter(label__exact=label_string, user=self.user).delete()
 
+        # remove the same label from all assigned resources
+        self.remove_resource_label(this_label)
```

Comment:
I would actually prefer that these operations be kept separate and done at the next higher level. This would allow more flexibility later. However, for the purposes of the AGU release, this expedient is fine.